### PR TITLE
Assignment_1

### PR DESCRIPTION
**Pull request by lisafeeney33**
## Checklist
- [ ] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by tianyi21 — CHANGES_REQUESTED** [review]
Hi Lisa!

Thank you for submitting your DL assignments.

I noticed the following issue(s) with your submission. I would appreciate it if you could address them by posting a comment below. There is no need to revise your submission.
* Could you please provide a short justification for your choices in Task 3a and Task 3b? I know the design choices might be simple and straightforward here, but it's required in the problem statement.

If you have any questions, please don't hesitate to let us know.

Thanks,
Tianyi [LS]

**Comment by lisafeeney33** [comment]
Hello - Sorry for missing that part of the question!  Here is the requested information:

Justification for choices in 3A (Explain your choice of the number of epochs used)

  *
10 is a modest number of epochs and is a good starting point in experimentation.
  *
It enables the model to learn patterns without spending excessive time if the number is too high
  *
Selecting a higher number of epochs could lead to overfitting (effectively the model memorizes the training data but then performs poorly on unseen data)
  *
Training for 10 epochs balances computational efficiency versus providing the model enough time to learn

Justification for choices in 3B (Report at least one other metric, and explain what it measures and how it is calculated.)

  *
I selected Mean Intersection over Union (IoU)
     *
First step: Compute IoU for each class

[cid:cde30931-1783-40f6-9810-8683e122e369]

        *
IoU measures the overlap between the predicted and true regions of a particular class.
        *
AoO = area where the predicted and ground truth segments agree
        *
AoU = combined area covered by the predicted and ground truth segments
     *
Second step: calculate Mean IoU
        *
average IoU across all classes
        *
evaluates the performance of segmentation models
  *
Mean IoU: 0.2705
     *
Interpretation = on average, ~27.05% of the predicted and ground truth regions overlap across all classes
     *
A Mean IoU of less than 5 indicates that the model is struggling to accurately segment the input

________________________________

**Review by tianyi21 — APPROVED** [review]
Hi Lisa,

Thank you for submitting your DL assignments. Your assignment 1 is now complete.

Tianyi [LS]
